overlord/snapstate/aliasesv2.go

```diff
@@ -343,6 +347,27 @@ func checkAliasesConflicts(st *state.State, snapName string, candAutoDisabled bo
 	return nil, nil
 }
 
+// checkSnapAliasConflict checks whether snapName and its command
+// namepsace conflicts against installed snap aliases.
+func checkSnapAliasConflict(st *state.State, snapName string) error {
+	prefix := fmt.Sprintf("%s.", snapName)
+	snapStates, err := All(st)
+	if err != nil {
+		return err
+	}
+	for otherSnap, snapst := range snapStates {
+		autoDisabled := snapst.AutoAliasesDisabled
+		for alias, target := range snapst.Aliases {
+			if alias == snapName || strings.HasPrefix(alias, prefix) {
+				if target.Effective(autoDisabled) != "" {
```

Comment:
if we don't check whether aliases are really on on the other side, we stop installing the snap even if the other one was installed unaliased, that seems more controlled but too harsh